File: 2433-find-the-original-array-of-prefix-xor/2433-find-the-original-array-of-prefix-xor.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
#include <stdio.h>
#include <stdlib.h>
int* findArray(int* pref, int prefSize, int* returnSize){
    int* result = (int*)malloc(prefSize * sizeof(int));
    result[0] = pref[0];
    for (int i = 1; i < prefSize; i++) {
        result[i] = pref[i] ^ pref[i - 1];
    }
    *returnSize = prefSize;
    return result;
}

File: 2433-find-the-original-array-of-prefix-xor/2433-find-the-original-array-of-prefix-xor.py
class Solution:
    def findArray(self, pref: List[int]) -> List[int]:
        n = len(pref)
        result=[]
        result.append(pref[0])

        for i in range(1, n):
            result.append( pref[i] ^ pref[i - 1])

        return result
            